### ISSUE TITLE
I need to use the eg3d method on my own dataset

### ISSUE THREAD

**Issue by yanre-hyd** [oneThousand1000/EG3D-projector#20]
Hello, first of all, thank you very much for your work! I need to use the eg3d method on my own dataset recently. In your method, I saw that I need to run the run_projector.py program to obtain the 'first_inv' late code, but I found that it uses synthesized images in w_plus_projector.py? I used it to generate the first_inv '' late code, and in the final generation, I found that it also had an impact on the 3D effect generated by my own image, resulting in poor results. If I only want to generate a 3D effect from my own image, how can I make changes? What is the purpose of the 'first_inv' late code?

**Comment by oneThousand1000** [comment]
Please refer to https://arxiv.org/abs/2106.05744 for the function of first inv, which is actually the 'pivot'.

**Comment by yanre-hyd** [comment]
Thank you very much. I will take a look.

**Comment by yanre-hyd** [comment]
Also, regarding the result of generating 3D on my own image, the facial effect is very poor. How can I solve this problem？

**Comment by ChangYuance** [comment]
> Also, regarding the result of generating 3D on my own image, the facial effect is very poor. How can I solve this problem？

Have you solve this problem？ Is the method can use for own dataset？ I am very glad to see your response！Thank you thank you and thank you.

**Comment by yanre-hyd** [comment]
> > Also, regarding the result of generating 3D on my own image, the facial effect is very poor. How can I solve this problem？
> 
> Have you solve this problem？ Is the method can use for own dataset？ I am very glad to see your response！Thank you thank you and thank you.
Yes, This method is one of the commonly used data set image inversion methods.

**Comment by yanre-hyd** [comment]
> > Also, regarding the result of generating 3D on my own image, the facial effect is very poor. How can I solve this problem？
> 
> Have you solve this problem？ Is the method can use for own dataset？ I am very glad to see your response！Thank you thank you and thank you.

Yes, we do! This method is one of the commonly used data set image inversion methods.

**Comment by oneThousand1000** [comment]
Hi @ChangYuance @yanre-hyd , this project is a simple implementation of PTI 3DGAN inversion. It does have the limitations of poor quality, weak identity preservation, and so on. I recommend using SOTA 3DGAN inversion methods for better quality, e.g. : 

https://github.com/cvlab-kaist/3DGAN-Inversion
https://github.com/FeiiYin/SPI
https://github.com/jiaxinxie97/HFGI3D
https://github.com/XChenZ/invertAvatar

**Comment by ChangYuance** [comment]
thank you very much for your sharing. You are doing a great job!



---Original---
From: "Yiqian ***@***.***&gt;
Date: Thu, Apr 10, 2025 17:06 PM
To: ***@***.***&gt;;
Cc: "Chang ***@***.******@***.***&gt;;
Subject: Re: [oneThousand1000/EG3D-projector] I need to use the eg3d method onmy own dataset (Issue #20)


   
Hi @ChangYuance @yanre-hyd , this project is a simple implementation of PTI 3DGAN inversion. It does have the limitations of poor quality, weak identity preservation, and so on. I recommend using SOTA 3DGAN inversion methods for better quality, e.g. :
 
https://github.com/cvlab-kaist/3DGAN-Inversion
 https://github.com/FeiiYin/SPI
 https://github.com/jiaxinxie97/HFGI3D
 https://github.com/XChenZ/invertAvatar

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***&gt;
  oneThousand1000 left a comment (oneThousand1000/EG3D-projector#20)
 
Hi @ChangYuance @yanre-hyd , this project is a simple implementation of PTI 3DGAN inversion. It does have the limitations of poor quality, weak identity preservation, and so on. I recommend using SOTA 3DGAN inversion methods for better quality, e.g. :
 
https://github.com/cvlab-kaist/3DGAN-Inversion
 https://github.com/FeiiYin/SPI
 https://github.com/jiaxinxie97/HFGI3D
 https://github.com/XChenZ/invertAvatar
